### PR TITLE
test(e2e): fixes for nightly pipeline

### PR DESCRIPTION
**Pull request by chriswldenyer**
We may need to tag 'nightly' images outside of nightly runs.
Added Jenkins config option to tag a successful build as 'nightly'.
if tests pass when run manually.
Allow e2e tests to be disabled from Jenkins config.
Removed some duplicated code.
Replace deprecated 'continue' keyword with 'reinstall'.

### CODE REVIEW

**Comment by chriswldenyer** [comment]
bors try

File: Jenkinsfile [hunk 1]
```diff
@@ -81,11 +81,8 @@ def getTestPlan() {
   if (params.e2e_continuous == true)  {
     return xray_continuous_testplan
   }
-  def causes = currentBuild.getBuildCauses()
-  for(cause in causes) {
-    if ("${cause}".contains("hudson.triggers.TimerTrigger\$TimerTriggerCause")) {
-      return xray_nightly_testplan
-    }
+  if (isTimed() == true) {
```
Comment (blaisedias) [inline Jenkinsfile:84]: should this also be `if (isTimed() == true || params.run_as_nightly == true) ` ?

Comment (chriswldenyer) [reply Jenkinsfile:84]: Only if we want to send the results to the nightly test plan. I thought about this and thought that it was inappropriate for this use case.

**Review by blaisedias — APPROVED** [review]

**Comment by chriswldenyer** [comment]
bors try

**Review by tiagolobocastro — APPROVED** [review]

**Comment by chriswldenyer** [comment]
bors merge

**Pull request merged**